src/tests/integration/test_foundry_prove.py

```diff
@@ -569,15 +569,68 @@ def test_foundry_resume_proof(
     assert_fail(test, single(prove_res))
 
 
-ALL_INIT_CODE_TESTS: Final = ('InitCodeTest.test_init()', 'InitCodeTest.testFail_init()')
+ALL_INIT_CODE_TESTS: Final = tuple((TEST_DATA_DIR / 'foundry-init-code').read_text().splitlines())
 
 
-@pytest.mark.parametrize('test', ALL_INIT_CODE_TESTS)
-def test_foundry_init_code(test: str, foundry: Foundry, bug_report: BugReport | None, no_use_booster: bool) -> None:
+@pytest.mark.parametrize('test_id', ALL_INIT_CODE_TESTS)
+def test_foundry_init_code(
+    test_id: str,
+    foundry: Foundry,
+    update_expected_output: bool,
+    bug_report: BugReport | None,
+    no_use_booster: bool,
+    server: KoreServer,
+) -> None:
     if no_use_booster:
         pytest.skip()
 
-    prove_res = foundry_prove(
+    test = 'ConstructorTest.run_constructor()'
+
+    prove_res_1 = foundry_prove(
+        foundry,
+        tests=[(test_id, None)],
+        prove_options=ProveOptions(
+            run_constructor=True,
+            bug_report=bug_report,
+        ),
+        rpc_options=RPCOptions(
+            smt_timeout=300,
+            smt_retry_limit=10,
+            use_booster=not no_use_booster,
+        ),
+    )
+
+    # Then
+    if test_id != test:
+        assert_pass(test_id, single(prove_res_1))
+        return
+    else:
+        assert_fail(test, single(prove_res_1))
+
+    # Continue checking non-test function
+    check_pending(foundry, test, [19, 20, 21])
+
+    foundry_step_node(
```

Comment:
I think the execution of `ConstructorTest.run_constructor()` terminates early due to `fail_fast` being turned on by default. You can pass `fail_fast=False` to the first `foundry_prove`:
```py
    prove_res_1 = foundry_prove(
        foundry,
        tests=[(test_id, None)],
        prove_options=ProveOptions(
            run_constructor=True,
            bug_report=bug_report,
            fail_fast=False,
        ),
        rpc_options=RPCOptions(
            smt_timeout=300,
            smt_retry_limit=10,
            use_booster=not no_use_booster,
        ),
    )
```
which shouldn't affect other `init_code` tests since they are passing. The test can then look like this:
```py
@pytest.mark.parametrize('test_id', ALL_INIT_CODE_TESTS)
def test_foundry_init_code(
    test_id: str,
    foundry: Foundry,
    update_expected_output: bool,
    bug_report: BugReport | None,
    no_use_booster: bool,
    server: KoreServer,
) -> None:
    if no_use_booster:
        pytest.skip()

    test = 'ConstructorTest.run_constructor()'

    prove_res_1 = foundry_prove(
        foundry,
        tests=[(test_id, None)],
        prove_options=ProveOptions(
            run_constructor=True,
            bug_report=bug_report,
            fail_fast=False
        ),
        rpc_options=RPCOptions(
            smt_timeout=300,
            smt_retry_limit=10,
            use_booster=not no_use_booster,
        ),
    )

    # Then
    if test_id != test:
        assert_pass(test_id, single(prove_res_1))
        return
    else:
        assert_fail(test, single(prove_res_1))

    # And when
        show_res = foundry_show(
            foundry,
            test=test,
            to_module=True,
            sort_collections=True,
            omit_unstable_output=True,
            pending=True,
            failing=True,
            failure_info=True,
            counterexample_info=True,
            port=server.port,
        )

        # Then
        assert_or_update_show_output(show_res, TEST_DATA_DIR / f'show/{test}.expected', update=update_expected_output)
```

Reply:
That would requite updating expected output, although it's almost the same.